requirements-dev.txt

```diff
@@ -5,6 +5,7 @@ pennylane<0.37
 pyquil>=4.4,<4.11; python_version < "3.12"
 pytket>=1.26,<1.30
 qiskit>=0.44,<1.2
+azure-quantum
```

Comment:
Move under `# optional runtime dependencies` section and add both azure-quantum and azure-storage-blob deps with min / max version bounds, same as you gave it `pyproject.toml`.